### PR TITLE
Implement LambdaParameterNaming

### PR DESCRIPTION
**Pull request by BraisGabin**
This implements part of #4131

### CODE REVIEW

File: detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/NamingRulesSpec.kt [hunk 1]
```diff
@@ -13,7 +13,6 @@ class NamingRulesSpec : Spek({
                 data class D(val i: Int, val j: Int)
                 fun doStuff() {
                     val (_, HOLY_GRAIL) = D(5, 4)
-                    emptyMap<String, String>().forEach { _, V -> println(V) }
```
Comment (BraisGabin) [inline detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/NamingRulesSpec.kt:16]: Why was this here? Should we support this? Is there any documentation that talks about this case?

Comment (chao2zhang) [reply detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/NamingRulesSpec.kt:16]: Is the problem about why we allow the capital `V` by default?

It feels like this is a test case that verifying that we do not have any naming requirement for lambda parameters 🙈 . I think your change makes sense.

**Review by chao2zhang — APPROVED** [review]

File: detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/NamingRulesSpec.kt (hunk 1 above)
Comment (chao2zhang) [reply detekt-rules-naming/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/NamingRulesSpec.kt:16]: Is the problem about why we allow the capital V by default?

It feels like this is a test case that verifying that we do not have any naming requirement for lambda parameters 🙈 . I think your change makes sense.

File: detekt-rules-naming/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/LambdaParameterNaming.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        Severity.Style,
+        "Lambda parameter names should follow the naming convention set in the projects configuration.",
+        debt = Debt.FIVE_MINS
+    )
+
+    @Configuration("naming pattern")
+    private val parameterPattern: Regex by config("[a-z][A-Za-z0-9]*", String::toRegex)
+
+    override fun visitParameter(parameter: KtParameter) {
+        val parameters: List<KtNamedDeclaration> = parameter.destructuringDeclaration?.entries ?: listOf(parameter)
```
Comment (marschwar) [inline detekt-rules-naming/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/LambdaParameterNaming.kt:31]: As far as I can tell there is nothing restricting this to lambda expressions.

Should this fail or pass?

```kotlin
it("ignores invalid function parameters") {
    val code = """
        fun foo(BAR: String) = Unit
    """
    assertThat(LambdaParameterNaming().compileAndLint(code))
        .isEmpty()
}
```

Comment (BraisGabin) [reply detekt-rules-naming/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/LambdaParameterNaming.kt:31]: You are right, there is nothing restrictin this to lambda. But all the naming rules are kind of strange. They aren't called by the normal rule engine. They are called from `detekt-rules-naming/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/NamingRules.kt`. And it only cal this function when the parameter is from a lambda. You can check my changes in that file.
